hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -615,6 +607,129 @@ def _get_top_data(topfile):
+                    raise CommandExecutionError('File data is not formed as a dict {0}'
+                                                .format(f_data))
+                conf_data = _dict_update(conf_data,
+                                        f_data,
+                                        recursive_update=True,
+                                        merge_lists=True)
+    if conf_data:
+        try:
+            log.debug("Writing config to osquery.conf file")
+            with open(configfile, "w") as cf:
+                json.dump(conf_data, cf)
+        except Exception as e:
+            log.error("Failed to generate osquery conf file using topfile {0}".format(e))
+    
+    return configfile
+
+
+def _generate_osquery_flags_file(flagstopfile):
+    '''
+    Function to dynamically create osquery flags file.
+    This function would load osquery flags in YAML format and make use of topfile to selectively load file(s) based on grains
+    '''
+
+    log.info("Generating osquery flags file using topfile: {0}".format(flagstopfile))
+    saltenv = __salt__['config.get']('hubblestack:nova:saltenv', 'base')
+    osqueryd_path = 'salt://hubblestack_nebula_v2'
+    cached = __salt__['cp.cache_dir'](osqueryd_path, saltenv=saltenv)
+    log.debug('Cached nebula files to cachedir')
+    cachedir = os.path.join(__opts__.get('cachedir'), 'files', saltenv, 'hubblestack_nebula_v2')
+    base_path = cachedir
+
+    osqd_flags = _get_osquery_top_data(flagstopfile)
+    flagfile = os.path.join(base_path, "osquery.flags")
+    flags_data = {}
+    osqd_flags = ['salt://hubblestack_nebula_v2/' + config.replace('.', '/') + '.yaml' for config in osqd_flags]
+    for fh in osqd_flags:
+        if 'salt://' in fh:
+            orig_fh = fh
+            fh = __salt__['cp.cache_file'](fh)
+        if fh is None:
+            log.error('Could not find file {0}.'.format(orig_fh))
+            return None
+        if os.path.isfile(fh):
+            with open(fh, 'r') as f:
+                f_data = yaml.safe_load(f)
+                if not isinstance(f_data, dict):
+                    raise CommandExecutionError('File data is not formed as a dict {0}'
+                                                .format(f_data))
+                flags_data = _dict_update(flags_data,
+                                        f_data,
+                                        recursive_update=True,
+                                        merge_lists=True)
+    if flags_data:
+        try:
+            log.debug("Writing config to osquery.flags file")
+            with open(flagfile, "w") as cf:
+                for key in flags_data:
+                    propdata = "--" + key + "=" + flags_data.get(key) + "\n"
+                    cf.write(propdata)
+        except Exception as e:
+            log.error("Failed to generate osquery flags file using topfile {0}".format(e))
+    
+    return flagfile
+
+
+def _get_osquery_top_data(topfile):
```

Comment:
Can we re-use (might need slight modifications) the existing `_get_top_data()` function? Seems like the topfiles are similar enough we can make it work without so much duplication.

Reply:
Done.